keps/sig-auth/20190806-serviceaccount-tokens.md

```diff
@@ -263,6 +264,35 @@ service account token on behalf of pods running on that node. The
 NodeRestriction admission controller will require that these tokens are pod
 bound.
 
+### Safe Adoption
+
+#### Blocker: In-cluster clients should reload fresh token
+
+- Summary: Bound service account tokens are time-bound. If in-cluster clients 
+do not load fresh token from projected volume, tokens would be rejected since 
+they are expired.
+
+- Rollout plan:
+  1. Modify admission-injected token source to request for token expiring at an 
+  uncommon value. (1 hour 10 seconds for example)
+  1. Modify kube apiserver TokenRequest API, providing an option 
+  `--mint-longer-token` to mint a token longer than requested if expiration 
+  requested is the uncommon value set in step i (+30 days for example), while 
```

Comment:
I would suggest choosing a time that exceeds the amount of time a given kubelet release is supported for (e.g. 1 year) to be safe (a workload running for >30 days seems very reasonable). When a kubelet is upgraded, it involves draining/restarting pods running on that node, which would cause them to pick up the new token. With pod-bound tokens (even ones with a long life) we still gain significant security benefits (the credential becomes invalid as soon as the associated pod is removed)

Reply:
Ack. Proposed 1 year.